packages/lib/services/userSegment.ts

```diff
@@ -0,0 +1,476 @@
+import { prisma } from "@formbricks/database";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/errors";
+import {
+  TActionMetric,
+  TAllOperators,
+  TBaseFilterGroup,
+  TUserSegmentActionFilter,
+  TUserSegmentAttributeFilter,
+  TUserSegmentConnector,
+  TUserSegmentDeviceFilter,
+  TUserSegmentSegmentFilter,
+  TUserSegmentUpdateInput,
+  ZUserSegmentFilterGroup,
+  isResourceFilter,
+} from "@formbricks/types/v1/userSegment";
+import {
+  getFirstOccurrenceDaysAgo,
+  getLastMonthEventCount,
+  getLastOccurrenceDaysAgo,
+  getLastQuarterEventCount,
+  getLastWeekEventCount,
+  getTotalOccurrences,
+} from "./actionsHelpers";
+
+export const createUserSegment = async (
+  environmentId: string,
+  surveyId: string,
+  title: string,
+  description: string,
+  isPrivate: boolean,
+  filters: TBaseFilterGroup
+) => {
+  const userSegment = await prisma.userSegment.create({
+    data: {
+      environmentId,
+      title,
+      description,
+      isPrivate,
+      filters,
+      ...(surveyId && {
+        surveys: {
+          connect: {
+            id: surveyId,
+          },
+        },
+      }),
+    },
+    include: {
+      surveys: {
+        select: {
+          id: true,
+        },
+      },
+    },
+  });
+
+  return userSegment;
+};
+
+export const getAllUserSegments = async (environmentId: string) => {
```

Comment:
please rename to `getUserSegments()` to stay with current naming of other services